Test/Models/ConcatenatedModel.cpp

```diff
@@ -219,108 +313,28 @@ BOOST_AUTO_TEST_CASE( CONCATENATED_MODEL_SERIALIZE )
 	oa << model;
 
 	//and create a new model from the serialization
-	Softmax netTest1;
-	Softmax netTest2;
-	ConcatenatedModel<RealVector,RealVector> modelDeserialized (&netTest1,&netTest2);
+	LinearModel<> netTest1;
+	NeuronLayer<SoftmaxNeuron<> > netTest2;
+	LinearModel<> netTest3;
+	ConcatenatedModel<RealVector> modelDeserialized  = netTest1 >> netTest2 >> netTest3;
 	istringstream inputStream(outputStream.str());  
 	TextInArchive ia(inputStream);
 	ia >> modelDeserialized;
 	
 	//test whether serialization works
 	//first simple parameter and topology check
 	BOOST_CHECK_SMALL(norm_2(modelDeserialized.parameterVector() - testParameters),1.e-50);
-	BOOST_REQUIRE_EQUAL(net1.inputSize(),netTest1.inputSize());
-	BOOST_REQUIRE_EQUAL(net2.inputSize(),netTest2.inputSize());
+	BOOST_REQUIRE_EQUAL(net1.inputShape(),netTest1.inputShape());
+	BOOST_REQUIRE_EQUAL(net1.outputShape(),netTest1.outputShape());
+	BOOST_REQUIRE_EQUAL(net2.inputShape(),netTest2.inputShape());
+	BOOST_REQUIRE_EQUAL(net2.outputShape(),netTest2.outputShape());
+	BOOST_REQUIRE_EQUAL(net3.inputShape(),netTest3.inputShape());
+	BOOST_REQUIRE_EQUAL(net3.outputShape(),netTest3.outputShape());
 	for (size_t i=0; i<1000; i++)
 	{
 		RealVector output = modelDeserialized(dataset.element(i).input);
 		BOOST_CHECK_SMALL(norm_2(output -dataset.element(i).label),1.e-50);
```

Comment:
"1.e-50" - for labels? With double precision, for all practical purposes, that's zero - right? (Unless we have really miniscule regression labels.)